sklearn/metrics/ranking.py

```diff
@@ -821,3 +821,251 @@ def label_ranking_loss(y_true, y_score, sample_weight=None):
+        or non-thresholded measure of decisions (as returned by
+        "decision_function" on some classifiers).
+
+    k : int, optional (default=None)
+        Only consider the highest k scores in the ranking. If None, use all
+        outputs.
+
+    log_basis : float, optional (default=2)
+        Basis of the logarithm used for the discount. A low value means a
+        sharper discount (top results are more important).
+
+    sample_weight : array-like of shape = [n_samples], optional, dtype = float.
+        Sample weights. If None, all samples are given the same weight.
+
+    Returns
+    -------
+    discounted_cumulative_gain : float
+        The averaged sample DCG scores.
+
+    References
+    ----------
+    `Wikipedia entry for Discounted Cumulative Gain
+        <https://en.wikipedia.org/wiki/Discounted_cumulative_gain>`_
+
+    Jarvelin, K., & Kekalainen, J. (2002).
+    Cumulated gain-based evaluation of IR techniques. ACM Transactions on
+    Information Systems (TOIS), 20(4), 422-446.
+
+    Wang, Y., Wang, L., Li, Y., He, D., Chen, W., & Liu, T. Y. (2013, May).
+    A theoretical analysis of NDCG ranking measures. In Proceedings of the 26th
+    Annual Conference on Learning Theory (COLT 2013)
+
+    See also
+    --------
+    ndcg_score :
+        The Discounted Cumulative Gain divided by the Ideal Discounted
+        Cumulative Gain (the DCG obtained for a perfect ranking), in order to
+        have a score between 0 and 1.
+
+    """
+    check_consistent_length(y_true, y_score, sample_weight)
+    return np.average(
+        _dcg_sample_scores(y_true, y_score, k=k, log_basis=log_basis),
+        weights=sample_weight)
+
+
+def _ndcg_sample_scores(y_true, y_score, k=None):
+    """Compute Normalized Discounted Cumulative Gain.
+
+    Sum the true scores ranked in the order induced by the predicted scores,
+    after applying a logarithmic discount. Then divide by the best possible
+    score (Ideal DCG, obtained for a perfect ranking) to obtain a score between
+    0 and 1.
+
+    This ranking metric yields a high value if true labels are ranked high by
+    ``y_score``.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples, n_labels], dtype = int or float
+        True targets of multilabel classification, or true scores of entities
+        to be ranked.
+
+    y_score : array, shape = [n_samples, n_labels], dtype = float
+        Target scores, can either be probability estimates, confidence values,
+        or non-thresholded measure of decisions (as returned by
+        "decision_function" on some classifiers).
+
+    k : int, optional (default=None)
+        Only consider the highest k scores in the ranking. If None, use all
+        outputs.
+
+    Returns
+    -------
+    normalized_discounted_cumulative_gain : array, shape = [n_samples,]
+        The NDCG score for each sample (float in [0., 1.]).
+
+    References
+    ----------
```

Comment:
These are internal functions, and docs won't be built for them, so I think you could remove the References section here and in  `_dcg_sample_scores` particularly since these references can be found in the corresponding public functions.